### PR TITLE
chore(ci): upload coredump

### PR DESCRIPTION
**Pull request by ADD-SP**
### Summary

Upload core dump if it exists, it is good for troubleshooting in the local env,

### Checklist

- [N/A] The Pull Request has tests
- [N/A] There's an entry in the CHANGELOG
- [N/A] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

N/A

### Issue reference

N/A

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -96,12 +99,23 @@ jobs:
...
+        echo "sysctl -w fs.suid_dumpable=2" >> run.sh
+        echo "sysctl -p" >> run.sh
+        echo "source ${{ env.BUILD_ROOT }}/kong-dev-venv.sh" >> run.sh
+        echo "TEST_CMD=\"bin/busted -v -o htest spec/01-unit\"" >> run.sh
+        echo "if [[ \$KONG_TEST_COVERAGE = true ]]; then" >> run.sh
+        echo "  TEST_CMD=\"\$TEST_CMD --coverage\"" >> run.sh
+        echo "fi" >> run.sh
+        echo "\$TEST_CMD" >> run.sh
+
+        sudo bash run.sh
```
Comment (fffonion) [inline .github/workflows/build_and_test.yml:118]: we can use sudo -E bin/busted -v -o htest spec/01-unit to passthrough env vars without need to >> run.sh

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -162,12 +191,20 @@ jobs:
...
+        name: build
+        path: .
+
+    - name: Extract the artifact contents
+      run: |
+        sudo tar -C /usr/local/ -xzvf artifacts/build.tar.gz
+        sudo mkdir -p ${{ env.BUILD_ROOT }}/kong-dev
+        sudo tar -C ${{ env.BUILD_ROOT }}/kong-dev -xzvf artifacts/build.tar.gz
+
+    - name: Install Kong dev
```
Comment (fffonion) [inline .github/workflows/build_and_test.yml:206]: are those `Install Kong dev` steps needed?

Comment (ADD-SP) [reply .github/workflows/build_and_test.yml:206]: @fffonion It will install some dev-rocks and some binary like `grpcurl`.

Comment (fffonion) [reply .github/workflows/build_and_test.yml:202]: iirc we already installed the dev rocks on build step for ce (and cached) 🤔

Comment (ADD-SP) [reply .github/workflows/build_and_test.yml:202]:
```
- name: Build Dev Kong dependencies
      if: steps.cache-deps.outputs.cache-hit != 'true'
      run: |
        make install-dev-rocks
```

This line has been removed in this PR, to keep consistency with EE.

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -270,12 +313,23 @@ jobs:
...
+      run: make dev LIBRARY_PREFIX=/usr/local/kong
+
+    - name: Install Kong dev
```
Comment (fffonion) [inline .github/workflows/build_and_test.yml:331]: do we need this? (and it's duplicated twice)

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -336,12 +400,23 @@ jobs:
...
+        name: build
+        path: .
+
+    - name: Extract the artifact contents
+      run: |
+        sudo tar -C /usr/local/ -xzvf artifacts/build.tar.gz
+        sudo mkdir -p ${{ env.BUILD_ROOT }}/kong-dev
+        sudo tar -C ${{ env.BUILD_ROOT }}/kong-dev -xzvf artifacts/build.tar.gz
+
+    - name: Install Kong dev
```
Comment (fffonion) [inline .github/workflows/build_and_test.yml:415]: do we still need this? (and it's duplicated twice)

**Comment by dndx** [comment]
Is this still needed?

**Pull request closed without merging**